doc/api/os.md

```diff
@@ -26,6 +26,17 @@ The operating system-specific end-of-line marker.
 * `\n` on POSIX
 * `\r\n` on Windows
 
+## `os.availableParallelism()`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {integer}
+
+Returns an estimate of the default amount of parallelism a program should use.
+Always returns a value greater than zero.
```

Comment:
Some explanation about how that estimate is reached would be helpful here.

Reply:
Here is the PR that added this functionality to libuv - https://github.com/libuv/libuv/commit/f250c6c73ee45aa93ec44133c9e0c635780ea741. If you have some exact text you'd like to see based on that, I'm happy to add it. The CI has been so flaky (and frustrating) that I don't want to go back and forth with pushing changes, running the CI, making more changes, etc. - I'd rather just ship whatever suggestions you have.

Alternatively, a docs update could be added after the fact since the CI requirements for docs only changes are much lower.

Reply:
The implementation of `uv_available_parallelism()` will likely evolve over time so the docs here shouldn't be too specific, or they'll become outdated.

My suggestions are to a) link to the libuv documentation, b) leave it out, or c) say we consult the animating spirits inside the computer.

Reply:
I'd like to see how option C plays out.

Reply:
```
Returns an estimate of the default amount of parallelism a program should use. Always returns a value greater than zero.

On Linux, inspects the calling thread’s CPU affinity mask to determine if it has been pinned to specific CPUs.

On Windows, the available parallelism may be underreported on systems with more than 64 logical CPUs.

On other platforms, reports the number of CPUs that the operating system considers to be online.

The heuristics used are expected to evolve over time and may vary across operating systems.
```

Or, if nothing else, link to the libuv documentation.

Reply:
@cjihrig ... I would suggest that a full CI run likely isn't necessary if these doc changes are made so long as it passes linting and basic github action checks.

Reply:
Does not answer the question of what it does inside of a cgroup, which is the primary problem with cpus()

Reply:
It's not mentioned because it's oblivious to the presence of cgroups.

Reply:
I'm not a fan of adding new features that replicate open bugs in existing functionality. 

Also [#28855 ](https://github.com/nodejs/node/issues/28855) specifically asks for this problem to be fixed, which this PR does not seem to accomplish?

Reply:
I'm lousy at reading C code, but I think this is saying:

https://github.com/libuv/libuv/commit/f250c6c73ee45aa93ec44133c9e0c635780ea741
https://www.cygwin.com/bugzilla/show_bug.cgi?id=27645

That availableParallelism() works on Alpine Linux (which is musl based?) but not on Ubuntu. But the extra call in the conditional block in the libuv commit above emulates the musl behavior. If I've followed that right, then the answer is 'yes' and it's *not* oblivious to cgroups.

Reply:
It's oblivious in the "cache oblivious algorithm" sense: libuv doesn't have to care whether cgroups are active or not; if the container is set up properly, it'll produce the right answer.